src/state_machine/signer/mod.rs

```diff
@@ -577,7 +577,22 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
             .map(|nr| nr.signer_id)
             .collect::<Vec<u32>>();
 
-        debug!("Got SignatureShareRequest for signer_ids {:?}", signer_ids);
+        let signer_id_set = sign_request
+            .nonce_responses
+            .iter()
+            .map(|nr| nr.signer_id)
+            .collect::<BTreeSet<u32>>();
+
+        if signer_ids.len() != signer_id_set.len()
+            || signer_id_set.is_empty()
+            || signer_id_set.len() > self.total_signers.try_into().unwrap()
+            || signer_id_set.last().unwrap() > &self.total_signers
+        {
+            warn!(signer_ids = ?signer_ids, "Got SignatureShareRequest with invalid NonceResponse");
```

Comment:
This can be
```rust
warn!(?signer_ids, "Got SignatureShareRequest with invalid NonceResponse");
```
same for the log line below.

Reply:
Done